### DISCUSSION TITLE
Add / Allow a user defined context/field in search object

### DISCUSSION THREAD

**Discussion by geminigeek** [meilisearch/product#761] [Feedback & Feature Proposal]
hi,

i am proposing a feature where we can set any arbitrary identifier/context object in the search object, then that context should be returned  in the search results, this will really help in multi_search cases and A/B testing, where the search object is coming from a DB or an api and not predefined 

useful for this feature https://www.meilisearch.com/docs/reference/api/multi_search

i am giving an example 

current search object
```
{
  q: query,
  indexUid: indexName

}
```

proposed search object 
```
{
  q: query,
  indexUid: indexName,
  context:{ anyKey:"value"}
}
```

proposed result

```
{
  "hits": [
    {
      "id": 2770,
      "title": "American Pie 2",
      "poster": "https://image.tmdb.org/t/p/w1280/q4LNgUnRfltxzp3gf1MAGiK5LhV.jpg",
      "overview": "The whole gang are back and as close as ever. They decide to get even closer by spending the summer together at a beach house. They decide to hold the biggest…",
      "release_date": 997405200
    },
],
"offset": 0,
"limit": 20,
"estimatedTotalHits": 976,
"processingTimeMs": 35,
"query": "query",
"context":{ anyKey:"value"}
```

**Comment by macraig** [comment]
Hi @geminigeek , can you give and example of how the "context" field would help your use case? What are you trying to achieve as a result?

**Reply by geminigeek** [reply]
hi,

i want to run multiple queries with 'multi_search' based on the original query string , but not always only some time ,  these query object comes from another api / function , i do not know how many queries can come or what those query are its by a logic (A/B test)
 
when the results come i want to prepare final result docs based on records pulled by all queries ,to prepare result i need to identify the query result by a key i send in context, currently only way to do is by order of the queries 

if there is a way to handle this case do let me know!

**Reply by macraig** [reply]
Hello again! We don't have a way of handling this case at the moment. However, we are working on a somewhat unrelated feature that will eventually have a "search with context" in it, which could serve your use case. We haven't prioritized this yet, but I can let you know when there's an update on it.

**Reply by geminigeek** [reply]
hi,

thanks for your reply, no issues if this feature is not planned , i am leaving this comment just in case its implemented, its fairly easy i am explaining a little below in coding terminology

1. destruct the query object remove the context key ( or what ever key we can call this) from query object.
2.  do the search like its done currently 
3. before sending the result add context key in search result object
4. done
